runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1066,6 +1513,14 @@ TR::Register *J9::Power::TreeEvaluator::awrtbariEvaluator(TR::Node *node, TR::Co
    bool needSync = (node->getSymbolReference()->getSymbol()->isSyncVolatile() && TR::Compiler->target.isSMP());
    bool lazyVolatile = false;
 
+   if (cg->comp()->getOption(TR_EnableFieldWatch) && !node->getSymbolReference()->getSymbol()->isArrayShadowSymbol())
+      {
+      TR::TreeEvaluator::rdWrtbarHelperForFieldWatch(node, cg, sideEffectRegister, valueReg);
+      // The Third child (sideEffectNode) and valueReg's node is also used by the store evaluator below.
+      // The store evaluator will also evaluate+decrement it. In order to avoid double
+      // decrementing the node we skip doing it here and let the store evaluator do it.
```

Comment:
Was this comment meant for the preceding function call? If so we should move it there.